Updated Albert model Card

souvikchand opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Updates `ALBERT` model card as per #[36979](https://github.com/huggingface/transformers/issues/36979)


## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)?  https://github.com/huggingface/transformers/issues/36979
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?
@stevhliu Please check the PR and see if it's alright :smile:


<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker
- vision models: @amyeroberts, @qubvel
- speech models: @eustlb
- graph models: @clefourrier

Library:

- flax: @gante and @Rocketknight1
- generate: @zucchini-nlp (visual-language models) or @gante (all others)
- pipelines: @Rocketknight1
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @zach-huggingface and @SunMarc
- chat templates: @Rocketknight1

Integrations:

- deepspeed: HF Trainer/Accelerate: @SunMarc @zach-huggingface
- ray/raytune: @richardliaw, @amogkam
- Big Model Inference: @SunMarc
- quantization (bitsandbytes, autogpt): @SunMarc @MekkCyber

Documentation: @stevhliu

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @Rocketknight1
- PyTorch: See Models above and tag the person corresponding to the modality of the example.
- TensorFlow: @Rocketknight1

 -->

souvikchand committed:
Updated Albert model Card

stevhliu commented on docs/source/en/model_doc/albert.md:
````diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
+    print(tokenizer.decode(outputs.logits[0].argmax(-1)))
+```
+
+> ALBERT is not compatible with `AttentionMaskVisualizer` as it uses masked self-attention rather than causal attention. So don't have `_update_causal_mask` method.
````
```suggestion
```

souvikchand replied:
Since almost every model's model card will have an `AttentionMaskVisualizer` but ALBERT is not getting it, isn't it good to explain why it don't have any?

stevhliu replied:
The `AttentionMaskVisualizer` has only been implemented for ~50/300 models so far so I think its okay to leave it out at the moment.

souvikchand replied:
ok alright. thanks for your response.

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
-[kamalkraj](https://huggingface.co/kamalkraj). The original code can be found [here](https://github.com/google-research/ALBERT).
+## Notes
 
+- All tokens attend to all others (like BERT).
```
```suggestion
- Inputs should be padded on the right because BERT uses absolute position embeddings.
- The embedding size `E` is different from the hidden size `H` because the embeddings are context independent (one embedding vector represents one token) and the hidden states are context dependent (one hidden state represents a sequence of tokens). The embedding matrix is also larger because `V x E` where `V` is the vocabulary size. As a result, it's more logical if `H >> E`. If `E < H`, the model has less parameters.
```

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
+## Notes
 
+- All tokens attend to all others (like BERT).
+- ALBERT supports a maximum sequence length of 512 tokens.
+- Cannot be used for autoregressive generation (unlike GPT)
+- ALBERT requires absolute positional embeddings, and it expects right-padding (i.e., pad tokens should be added at the end, not the beginning).
+- ALBERT uses token_type_ids, just like BERT. So you should indicate which token belongs to which segment (e.g., sentence A vs. sentence B) when doing tasks like question answering or sentence-pair classification.
+- ALBERT uses a different pretraining objective called Sentence Order Prediction (SOP) instead of Next Sentence Prediction (NSP), so fine-tuned models might behave slightly differently from BERT when modeling inter-sentence relationships.
```
```suggestion
```

souvikchand replied:
could you please briefly explain why it's better not to mention internal differences like **SOP** or `token_type_ids` in the model card? I want to make sure I follow the best style when contributing next time.

stevhliu replied:
This is more useful background knowledge versus practical tips for a developer using the model so thats why we don't need to really mention the pretraining objective etc.

souvikchand replied:
ok. thanks for your guidance.

stevhliu commented on docs/source/en/model_doc/albert.md:
````diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
 
-### Using Scaled Dot Product Attention (SDPA)
+```py 
+import torch
````
You can simplify the whole thing to:

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="fill-mask",
    model="albert-base-v2",
    torch_dtype=torch.float16,
    device=0
)
pipeline("Plants create [MASK] through a process known as photosynthesis.", top_k=5)
```

souvikchand replied:
resolved

stevhliu commented on docs/source/en/model_doc/albert.md:
````diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
+<hfoption id="AutoModel">
+
+```py
+import torch
````
This can also be simplified a bit without the comments and include SDPA:

```py
import torch
from transformers import AutoModelForMaskedLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("albert/albert-base-v2")
model = AutoModelForMaskedLM.from_pretrained(
    "albert/albert-base-v2",
    torch_dtype=torch.float16,
    attn_implementation="sdpa",
    device_map="auto"
)

prompt = "Plants create energy through a process known as [MASK]."
inputs = tokenizer(prompt, return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    mask_token_index = torch.where(inputs["input_ids"] == tokenizer.mask_token_id)[1]
    predictions = outputs.logits[0, mask_token_index]

top_k = torch.topk(predictions, k=5).indices.tolist()
for token_id in top_k[0]:
    print(f"Prediction: {tokenizer.decode([token_id])}")
```

souvikchand replied:
it is done

souvikchand replied:
instead of `.to("cuda")` i wrote `.to(model.device)` . I hope it's okay

stevhliu reviewed (commented):
Nice, thanks for your contribution!

souvikchand committed:
Update docs/source/en/model_doc/albert.md

added the quotes in <hfoption id="Pipeline">

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

updated checkpoints

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

changed !Tips description

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

updated text

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

updated transformer-cli implementation

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

changed text

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

removed repeated description

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update albert.md

removed lines

souvikchand committed:
Update albert.md

updated pipeline code

souvikchand committed:
Update albert.md

updated auto model code, removed quantization as model size is not large, removed the attention visualizer part

souvikchand committed:
Update docs/source/en/model_doc/albert.md

updated notes

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update albert.md

reduced a  repeating point in notes

stevhliu commented on docs/source/en/model_doc/albert.md:
````diff
@@ -14,99 +14,92 @@ rendered properly in your Markdown viewer.
-...
+
+</hfoption>
+<hfoption id="transformers-cli">
+
+```bash
+echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers-cli run --task fill-mask --model albert-base-v2 --device 0
````
Following #37657, lets also update the `transformers` command

```suggestion
<hfoption id="transformers CLI">

```bash
echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers run --task fill-mask --model albert-base-v2 --device 0
```

stevhliu reviewed (commented):
Thanks, just a few more minor details :)

souvikchand committed:
Update docs/source/en/model_doc/albert.md

updated transformer-CLI

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand committed:
Update docs/source/en/model_doc/albert.md

removed extra notes

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

souvikchand commented:
@stevhliu, I have done suggested changes. please review and let me know if it is ok 😄

stevhliu commented:
Hey, sorry for the delay! PR looks good to me, feel free to mark it as ready for review whenever you're ready 🙂

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_37753). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

stevhliu reviewed (approved):
Thanks and no problem, this looks good!

druvdub commented on docs/source/en/model_doc/albert.md:
```diff
@@ -14,100 +14,100 @@ rendered properly in your Markdown viewer.
+
+ALBERT uses absolute position embeddings (like BERT) so padding is applied at right. Size of embeddings is 128 While BERT uses 768. ALBERT can processes maximum 512 token at a time. 
+>>>>>>> 7ba1110083 (Update docs/source/en/model_doc/albert.md
)
```
Hey @stevhliu, I found some merge conflict artifacts that weren't removed during the PR merge. These lines are still showing in the docs. Should I open a new PR to clean these up?

![image](https://github.com/user-attachments/assets/7bd24d71-9d05-4751-8f71-65755ae6683a)
